src/natcap/invest/utils.py

```diff
@@ -43,6 +44,205 @@
 # leaves Projected CRS alone
 DEFAULT_OSR_AXIS_MAPPING_STRATEGY = osr.OAMS_TRADITIONAL_GIS_ORDER
 
+# the same unit registry instance should be shared across everything
+u = pint.UnitRegistry()
+# see https://pint.readthedocs.io/en/stable/defining.html
+# pint doesn't allow multiple base units for the same dimension
+# but you can define mutliple dimensionless units with []
+# https://github.com/hgrecco/pint/issues/1278
+u.define('currency = [value]')    # non-specific unit of value
+u.define('pixel = []')            # non-specific unit of area
+u.define('projection_unit = []')  # non-specific unit of length
+# used in coastal vulnerability, the DEM pixel values measure
+# elevation (length) but the specific units don't matter
+u.define('linear_unit_of_elevation = []')  # non-specific unit of length
+# add "us_survey_foot" on to the aliases because it's used in some rasters
+u.define('survey_foot = 1200 / 3937 * meter = sft = us_survey_foot')
+# Vitamin A in the crop production nutrient table is measured in IUs
+# A special unit in pharmacology that measures biologically active substances
+# May be converted to weight or volume, but conversion factors are specific
+# to the substance. I couldn't find a definition of its dimensionality.
+u.define('international_unit = [biologic_amount] = iu = IU')
+# Use u.none for unitless measurements
+u.define('none = []')
+
+WORKSPACE_SPEC = {
+    "name": "Workspace",
+    "about": (
+        "The folder where all intermediate and output files of the model "
+        "will be written.  If this folder does not exist, it will be "
+        "created."),
+    "type": "directory",
+    "contents": {},
+    "validation_options": {
+        "exists": False,
+        "permissions": "rwx",
+    }
+}
+
+SUFFIX_SPEC = {
+    "name": "File suffix",
+    "about": (
+        'A string that will be added to the end of all files '
+        'written to the workspace.'),
+    "type": "freestyle_string",
+    "required": False,
+    "validation_options": {
+        "regexp": {
+            "pattern": "[a-zA-Z0-9_-]*"
+        }
+    }
+}
+
+N_WORKERS_SPEC = {
+    "name": "Taskgraph n_workers parameter",
+    "about": (
+        "The n_workers parameter to provide to taskgraph. "
+        "-1 will cause all jobs to run synchronously. "
+        "0 will run all jobs in the same process, but scheduling will take "
+        "place asynchronously. Any other positive integer will cause that "
+        "many processes to be spawned to execute tasks."),
+    "type": "number",
+    "units": u.none,
+    "required": False,
+    "validation_options": {
+        "expression": "value >= -1"
+    }
+}
+
+AREA = {
```

Comment:
The actual terminology of `AREA` used for a polygon vector is not intuitive to me. I wonder if there is a better name for this... I see `FILL` a lot in the vector realm as a styling indicator.

Reply:
Removed `AREA` since it was only used a couple times and wasn't clear